provider/examples_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2017, Oracle and/or its affiliates. All rights reserved.
+
+package provider
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+const examples_test_state_file = "test_examples.tfstate"
+
+var examples_test_allowed_environment_variables = []string{
+	"PATH",
+	"TF_VAR_user_ocid",
+	"TF_VAR_tenancy_ocid",
+	"TF_VAR_fingerprint",
+	"TF_VAR_private_key_path",
+	"TF_VAR_private_key_password",
+	"TF_VAR_region",
+	"TF_VAR_compartment_ocid",
+	"TF_VAR_AD",
+	"TF_VAR_ssh_public_key",
+	"TF_VAR_ssh_private_key",
```

Comment:
We can probably get rid of the confusing and only used in one example `TF_VAR_ssh_public_key` and `TF_VAR_ssh_private_key` environment variables using terraforms [file](https://www.terraform.io/docs/configuration/interpolation.html#file-path-) function

Reply:
Probably, but let's do that in a later revision.